tests/utils/queryApiGateway.test.ts

```diff
@@ -161,16 +162,51 @@ describe('Querying the API Gateway', () => {
       API_URL: goodUrl,
     })
 
-    const data = await queryApiGateway(goodRequest)
+    await expect(queryApiGateway(goodRequest, goodUniqueNumber)).rejects.toThrow(fsErrorMessage)
 
-    expect(data).toStrictEqual(emptyResponse)
     expect(fetch).toHaveBeenCalledTimes(0)
     expect(loggerSpy).toHaveBeenCalledWith('error', expect.anything(), 'Read certificate error')
 
     // Restore env vars
     restore()
   })
 
+  it('handles mismatched unique number responses', async () => {
+    // Mock process.env
+    const restore = mockEnv({
+      API_URL: goodUrl,
+    })
+
+    const mismatchedUniqueNumber = 'abcde'
+    await expect(queryApiGateway(goodRequest, mismatchedUniqueNumber)).rejects.toThrow(
+      'Mismatched API response and Header unique number (12345 and abcde)',
+    )
+
+    expect(fetch).toHaveBeenCalledTimes(1)
+    expect(loggerSpy).toHaveBeenCalledWith('error', expect.anything(), 'Unexpected API gateway response')
+
+    // Restore env vars
+    restore()
+  })
+
+  it('handles null unique number responses', async () => {
```

Comment:
I'm missing something... on line 199, we're sending null as the unique number? If the unique number is null, shouldn't we not be sending that request at all? And if we do send null, wouldn't it actually be matched (we send uniqueNumber: null, we get back uniqueNumber: null, `apiData?.uniqueNumber === uniqueNumber`)?

Reply:
this is if we send a number and get back null, not if we send null!

Reply:
does this clarify?
```suggestion
  it('handles API response with a null unique number', async () => {
```